### PR TITLE
Fix link to object if it is exposed in `__all__`

### PR DESCRIPTION
**Pull request by nathanthorpe**
Related issue https://github.com/mitmproxy/pdoc/issues/669

Search through the possible sources of the module when looking in `all_modules.`
Also use the path of the matched object.

### CODE REVIEW

File: pdoc/render_helpers.py [hunk 1]
```diff
@@ -336,23 +336,27 @@ def linkify_repl(m: re.Match):
...
-            if qualname:
-                assert isinstance(doc, pdoc.doc.Module)
-                doc = doc.get(qualname)
-            target_exists_and_public = (
-                doc is not None and context["is_public"](doc).strip()
-            )
-            if target_exists_and_public:
-                assert doc is not None  # mypy
+            # Check again if the object was re-exposed
+            for module, qualname in possible_sources(
```
Comment (mhils) [inline pdoc/render_helpers.py:340]: I _think_ this should use `qualname_candidates` here.

**Review by mhils — COMMENTED** [review]
Thank you for the PR! I generally agree that #669 is something we should support. 👍 

That being said - we are reaching ugly levels of function complexity here, not sure yet how we can make things simpler. Worst case it just needs more docstrings. I will take a closer look when I have a few cycles. In the meantime, could you please 1) add a snapshot test and 2) a CHANGELOG entry? Thanks!

**Comment by nathanthorpe** [comment]
Thanks for looking into it! 
Ok I believe I've made the appropriate test case. Also added a changelog entry.

File: pdoc/render_helpers.py (hunk 1 above)
Comment (nathanthorpe) [reply pdoc/render_helpers.py:340]: `qualname` is a string in this case, so I don't think it should be `candidates`

**Comment by liborvaneksw** [comment]
Hi! I'm facing the same issue. Unfortunately, this PR seems a bit stale. Do you plan to continue with this? Do you need a hand?

**Comment by nathanthorpe** [comment]
This PR should be ready for merge

**Comment by nathanthorpe** [comment]
I just tested it and it looks great, thank you!

**Pull request merged**